tests/test_enterprise/test_signals.py

```diff
@@ -625,3 +626,70 @@ def test_update_catalog_with_existing_service_catalog(self, api_client_mock):
             enabled_course_modes=enterprise_catalog.enabled_course_modes,
             publish_audit_enrollment_urls=enterprise_catalog.publish_audit_enrollment_urls
         )
+
+    @mock.patch('enterprise.signals.EnterpriseCatalogApiClient')
+    def test_update_enterprise_query(self, api_client_mock):
+        """
+        Tests the update_enterprise_query post_save signal.
+
+        Creates an EnterpriseCatalogQuery instance and two separate EnterpriseCatalog
+        instances that are associated with the query. The query's content filter is then
+        updated to see if the changes are applied across both related catalogs. Additionally,
+        there is a test to see if the sync is sent to the EnterpriseCatalogApi service
+        which is done through the mock api client.
+        """
+        content_filter_1 = {
+            'content_type': 'course',
+        }
+
+        content_filter_2 = {
+            'content_type': 'course2',
+        }
+
+        sender = mock.Mock()  # this would be an EnterpriseCatalogQuery class
+
+        # represents instance of EnterpriseCatalogQuery
+        test_query = EnterpriseCatalogQueryFactory(
+            content_filter=content_filter_1
+        )
+        # represents instances of EnterpriseCustomerCatalog
+        enterprise_catalog_1 = EnterpriseCustomerCatalogFactory(
+            sync_enterprise_catalog_query=True,
+            enterprise_catalog_query = test_query
+        )
+        enterprise_catalog_2 = EnterpriseCustomerCatalogFactory(
+            sync_enterprise_catalog_query=True,
+            enterprise_catalog_query = test_query
+        )
+
+        test_query.content_filter = content_filter_2
+
+        # kwargs to satisfy parameter
+        kwargs = {
+            'update_fields': None,
+            'raw': False,
+            'signal': '<django.db.models.signals.ModelSignal object at test>',
+            'using': 'default',
+            'created': False,
+        }
+
+        update_enterprise_query(sender, test_query, **kwargs)  # calls post_save signal
+
+        self.assertEqual(enterprise_catalog_1.content_filter, enterprise_catalog_2.content_filter, content_filter_2)
+
+        enterprise_catalog_1.save()
+        enterprise_catalog_2.save()
```

Comment:
Hm this seems like the functionality might not work quite right. We want the updates to get synced to the enterprise catalog service without having to manually save all of them